test/test_helper.rb

```diff
@@ -89,9 +89,20 @@ def after_teardown
   end
 end
 
+module TestCaseNetworkStubExtensions
+  def setup
+    Resolv::DNS.any_instance.stubs(:getname).returns("stubbed-host.example.com")
+    Resolv::DNS.any_instance.stubs(:getnames).returns(["stubbed-host.example.com"])
+    Resolv::DNS.any_instance.stubs(:getaddress).returns("127.0.0.13")
+    Resolv::DNS.any_instance.stubs(:getaddresses).returns(["127.0.0.13"])
```

Comment:
We shouldn't pretend it exists:
```
[6] pry(main)> Resolv.getaddresses('doesnotexist.example.com')
=> []
```

```suggestion
    Resolv::DNS.any_instance.stubs(:getaddresses).returns([])
```

Reply:
Ok ok, I will do exceptions. I will do this in a separate PR tho, I want this one to be backported.